### ISSUE TITLE
fadvise/drop fails on alpha

### ISSUE THREAD

**Issue by glaubitz** [util-linux/util-linux#3331]
The test `fadvise/drop` fails on alpha with util-linux 2.40.2:

```
glaubitz@debian-alpha:~/util-linux$ ./tests/ts/fadvise/drop
      fadvise: drop page caches related to a file ... FAILED (fadvise/drop)
glaubitz@debian-alpha:~/util-linux$
```

The contents of `./tests/output/fadvise/drop` are:

```
glaubitz@debian-alpha:~/util-linux$ cat ./tests/output/fadvise/drop
  RES  SIZE FILE
 512K  512K ddtest

whole file
status: 0
RES  SIZE FILE
 0B  512K ddtest

offset: 131072
status: 0
  RES  SIZE FILE
 128K  512K ddtest

length: 262144
status: 0
  RES  SIZE FILE
 256K  512K ddtest

offset: 131072, length: 262144 fd: 42
status: 0
  RES  SIZE FILE
 256K  512K ddtest

glaubitz@debian-alpha:~/util-linux$
```

**Comment by t-8ch** [comment]
Looks good, except maybe a trailing newline.
Can you post the diff?

**Comment by glaubitz** [comment]
Hmm, the diff seems to indicate a missing `ddtest` binary?

```diff
glaubitz@debian-alpha:~/util-linux$ cat ./tests/diff/fadvise/drop.err 
--- /dev/null   2024-12-23 23:57:09.504996407 +0100
+++ /home/glaubitz/util-linux/tests/output/fadvise/drop.err     2024-12-24 00:28:21.960112074 +0100
@@ -0,0 +1,5 @@
+fincore: failed to do cachestat: ddtest: No such file or directory
+fincore: failed to do cachestat: ddtest: No such file or directory
+fincore: failed to do cachestat: ddtest: No such file or directory
+fincore: failed to do cachestat: ddtest: No such file or directory
+fincore: failed to do cachestat: ddtest: No such file or directory
glaubitz@debian-alpha:~/util-linux$
```

**Comment by glaubitz** [comment]
So, the problem here seems to be the `fincore` command:

```
glaubitz@debian-alpha:~/util-linux$ dd if=/dev/zero of=ddtest bs=65536 count=8 conv=fsync >& /dev/null
glaubitz@debian-alpha:~/util-linux$ ls -l ddtest
-rw-rw-r-- 1 glaubitz glaubitz 524288 Dec 25 15:37 ddtest
glaubitz@debian-alpha:~/util-linux$ ./fincore -o RES,SIZE,FILE ddtest
fincore: failed to do cachestat: ddtest: No such file or directory
  RES  SIZE FILE
 512K  512K ddtest
glaubitz@debian-alpha:~/util-linux$
```

**Comment by t-8ch** [comment]
`misc-utils/fincore.c` falls back to `#define SYS_cachestat 451`, but on alpha it should actually be `561`.
Is the fallback used in your case?

Maybe we should drop the fallback altogether. (And use `__NR_cachestat` instead?)

**Comment by glaubitz** [comment]
Yep, already saw that. I hacked it to use `561` on alpha and it actually fixes this issue.

**Comment by glaubitz** [comment]
And this also fixes #3333.
